packages/@aws-cdk/aws-lambda/lib/function.ts

```diff
@@ -1030,31 +1041,46 @@ Environment variables can be marked for removal when used in Lambda@Edge by sett
     };
   }
 
+  private isQueue(deadLetterQueue: sqs.IQueue | sns.ITopic): deadLetterQueue is sqs.IQueue {
+    return (<sqs.IQueue>deadLetterQueue).queueArn !== undefined;
+  }
+
   private buildDeadLetterQueue(props: FunctionProps) {
     if (props.deadLetterQueue && props.deadLetterQueueEnabled === false) {
       throw Error('deadLetterQueue defined but deadLetterQueueEnabled explicitly set to false');
     }
 
-    if (!props.deadLetterQueue && !props.deadLetterQueueEnabled) {
+    let deadLetterQueue: sqs.IQueue | sns.ITopic;
+
+    if (!props.deadLetterQueue && !props.deadLetterQueueEnabled && !props.deadLetterTopic) {
       return undefined;
+    } else if (props.deadLetterTopic && props.deadLetterQueue) {
+      throw new Error('deadLetterQueue and deadLetterTopic cannot be specified together at the same time');
+    } else if (props.deadLetterTopic && (props.deadLetterQueueEnabled === false || props.deadLetterQueueEnabled === true)) {
+      throw new Error('deadLetterQueueEnabled and deadLetterTopic cannot be specified together at the same time');
```

Comment:
these conditions can be combined since they throw the same error - i think its fine to say `'deadLetterQueue'` for `deadLetterQueueEnabled`.

Reply:
Error is not exactly the same but I think I see your point 👍